### PR TITLE
Add reCaptcha enterprise support

### PR DESCRIPTION
**Pull request by AwesomeNipun**
Sending reCaptchaType string value in the request header for the login flow.

Screenshot 2023-06-08 at 14 30 57

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 1]
```diff
@@ -804,7 +808,9 @@ private String getCaptchaParams(String tenantDomain, int failedLoginAttempts) {
                         // SSO Login Captcha Config
                         if (Boolean.parseBoolean(connectorConfig.getValue()) ||
                                 forcefullyEnabledRecaptchaForAllTenants) {
-                            captchaParams = BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
+                            captchaParams = BasicAuthenticatorConstants.RECAPTCHA_PARAM + reCaptchaEnabled;
+                            captchaParams += BasicAuthenticatorConstants.RECAPTCHA_ENTERPRISE_PARAM +
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:812]: Do this for only reCaptchaEnterpriseEnabled

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 2]
```diff
@@ -880,18 +888,33 @@ private Properties getCaptchaConfigs() {
...
+                        log.debug("Empty values found for the captcha properties in the file " + CaptchaConstants
+                                .CAPTCHA_CONFIG_FILE_NAME + ".");
+                    }
+                    properties.clear();
+                }
+                if (reCaptchaEnterpriseEnabled) {
+                    if (StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_PROJECT_ID))) {
+
+                        if (log.isDebugEnabled()) {
+                            log.debug("Empty values found for the captcha properties in the file " + CaptchaConstants
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:912]: Improve the debug message

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 3]
```diff
@@ -804,7 +808,9 @@ private String getCaptchaParams(String tenantDomain, int failedLoginAttempts) {
                         // SSO Login Captcha Config
                         if (Boolean.parseBoolean(connectorConfig.getValue()) ||
                                 forcefullyEnabledRecaptchaForAllTenants) {
-                            captchaParams = BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
+                            captchaParams = BasicAuthenticatorConstants.RECAPTCHA_PARAM + reCaptchaEnabled;
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:811]: Better to hardcode the value

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 4]
```diff
@@ -880,18 +888,33 @@ private Properties getCaptchaConfigs() {
...
+                        StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_API_URL)) ||
+                        StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_SECRET_KEY)) ||
+                        StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_VERIFY_URL))) {
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("Empty values found for the captcha properties in the file " + CaptchaConstants
+                                .CAPTCHA_CONFIG_FILE_NAME + ".");
+                    }
+                    properties.clear();
+                }
```
Comment (AwesomeNipun) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:907]: can use a single if condition

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 5]
```diff
@@ -881,11 +889,15 @@ private Properties getCaptchaConfigs() {
         Properties properties = BasicAuthenticatorDataHolder.getInstance().getRecaptchaConfigs();
 
         if (properties != null && !properties.isEmpty() &&
-                Boolean.valueOf(properties.getProperty(CaptchaConstants.RE_CAPTCHA_ENABLED))) {
+                Boolean.parseBoolean(properties.getProperty(CaptchaConstants.RE_CAPTCHA_ENABLED))) {
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:892]: Is there a reason to edit this?

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 6]
```diff
@@ -881,11 +889,15 @@ private Properties getCaptchaConfigs() {
...
+
             if (StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_SITE_KEY)) ||
                     StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_API_URL)) ||
                     StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_SECRET_KEY)) ||
-                    StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_VERIFY_URL))) {
+                    StringUtils.isBlank(properties.getProperty(CaptchaConstants.RE_CAPTCHA_VERIFY_URL)) ||
+                    (properties.getProperty(CaptchaConstants.RE_CAPTCHA_TYPE).
```
Comment (hwupathum) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:898]: use `CONSTANT_STRING.equals(variableString)` not the other way around

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java (hunk 5 above)
Comment (AwesomeNipun) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java:892]: There was a suggestion in the IntelliJ Idea on this. Will revert back to the previous one (Boolean.valueOf) if this is not recommended.
Screenshot 2023-06-29 at 12 08 26

**Comment by Lakshan-Banneheke** [comment]
This PR is no longer required

**Pull request closed without merging**